test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/StrictCheckSpanProcessor.java

```diff
@@ -27,58 +26,39 @@ public StrictCheckSpanProcessor() {
 
     }
 
+    private static Map<String, MockSpanData> finishedSpanItems = new ConcurrentHashMap<>();
+
     @Override
     public void onStart(Span span) {
         spanMap.put(span.getSpanId(), Thread.currentThread().getStackTrace());
+        MockSpanData spanData;
+        if (span.getParentSpan() == null){
+            spanData = new MockSpanData(span.getSpanId(),
+                "0000000000", span.getTraceId(), System.nanoTime(), false, span.getSpanName());
+        } else {
+            spanData = new MockSpanData(span.getSpanId(),
+                span.getParentSpan().getSpanId(), span.getTraceId(), System.nanoTime(), false, span.getSpanName());
+        }
+        MockSpanData put = finishedSpanItems.put(span.getSpanId(), spanData);
     }
 
     @Override
     public void onEnd(Span span) {
         spanMap.remove(span.getSpanId());
-    }
-
-    /**
-     * Ensures that all the spans are closed. Throws exception message with stack trace of the method form
-     * where the span was created. We can enhance it to print all the failed spans in a single go based on
-     * the usability.
-     */
-    public void ensureAllSpansAreClosed() {
```

Comment:
Have we moved to somewhere else?

Reply:
Yes this will be covered as part of validator AllSpansAreEndedProperly